src/components/AmountInput.tsx

```diff
@@ -1,32 +1,43 @@
-import { useState } from 'react';
+import { forwardRef, useState } from 'react';
 import { useAtomValue } from 'jotai';
 import { displayFunctionsAtom } from 'store/app';
 import { stringifyValue, parseAsValue } from '@agoric/ui-components';
 import { AssetKind } from '@agoric/ertp';
-import type { Brand } from '@agoric/ertp/src/types';
 import StyledInput from './StyledInput';
+import type { Brand } from '@agoric/ertp/src/types';
+import type { Ref } from 'react';
+import clsx from 'clsx';
 
 type Props = {
   value?: bigint | null;
   label?: string;
   error?: string;
   brand?: Brand | null;
   suffix?: string;
+  actionLabel?: string;
   onChange?: (value: bigint) => void;
+  onAction?: () => void;
   disabled?: boolean;
 };
 
-const AmountInput = ({
-  value,
-  brand,
-  label,
-  error,
-  suffix,
-  onChange = () => {
-    /* noop */
-  },
-  disabled = false,
-}: Props) => {
+const AmountInput = (
+  {
+    value,
+    brand,
+    label,
+    error,
+    suffix,
+    actionLabel,
+    onChange = () => {
+      /* noop */
+    },
```

Comment:
consider defining `noop` or importing it from Lodash.
```suggestion
    onChange = noop,
```

Reply:
Defined above.